src/Sulu/Component/Content/Mapper/ContentMapper.php

```diff
@@ -1204,6 +1205,53 @@ private function createSql2Query($sql2, $limit = null)
         return $query;
     }
 
+    public function loadShallowStructureByNode(NodeInterface $contentNode, $localization, $webspaceKey)
+    {
+        $structureType = $this->nodeHelper->getStructureTypeForNode($contentNode) ? : Structure::TYPE_PAGE;
+        $propertyTranslator = $this->createPropertyTranslator($localization, $structureType);
+
+        $nodeType = $contentNode->getPropertyValueWithDefault(
+            $propertyTranslator->getName('nodeType'),
+            Structure::NODE_TYPE_CONTENT
+        );
+
+        $originTemplateKey = $this->defaultTemplates[$structureType];
+        $templateKey = $contentNode->getPropertyValueWithDefault(
+            $propertyTranslator->getName('template'),
+            $originTemplateKey
+        );
+
+        $templateKey = $this->templateResolver->resolve($nodeType, $templateKey);
+        $structure = $this->getStructure($templateKey, $structureType);
+
+        $structure->setUuid($contentNode->getPropertyValue('jcr:uuid'));
+        $structure->setNodeType(
+            $contentNode->getPropertyValueWithDefault(
+                $propertyTranslator->getName('nodeType'),
+                Structure::NODE_TYPE_CONTENT
+            )
+        );
+        $structure->setWebspaceKey($webspaceKey);
```

Comment:
So you have basically duplicated this code? Why not put it into a separate function, and call it at the other place as well?